### PR TITLE
Modernise project relationship page

### PR DESCRIPTION
**Pull request by timja**
This was the only usage in core of the long deprecated `f:editableComboBox` 
https://github.com/jenkinsci/jenkins/blob/c24633c303f1c2f339cdd6d29ab7c032acb79837/core/src/main/resources/lib/form/editableComboBox.jelly#L29

Few more around outside of core though: https://github.com/search?q=org%3Ajenkinsci%20editableComboBox&type=code

This cleans up the styling so the page doesn't look so out of place
I don't fully understand the page and so I haven't improved the rendering of the results, but the inputs and general look and feel are improved

Before:

image

After:

image

We could possibly get rid of the help sub-page but the content seemed a bit much so I just copied some of it over to explain the page a bit.

### Testing done

see screenshots above, I've also tested with missing parameters.

To setup the test environment I used an upstream and downstream project, created a file in the upstream, archived it (and enabled fingerprinting), enabled the copy artifact setting on the upstream job so that downstream could copy it and then copied the artifact in the downstream job recording the fingerprint (using copy artifact plugin)

### Proposed changelog entries

- Modernize project relationship page.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/java/jenkins/model/Jenkins.java
Comment (daniel-beck) [inline core/src/main/java/jenkins/model/Jenkins.java:1912]:
```suggestion
    @Restricted(DoNotUse.class)
    public ComboBoxModel doFillJobNameItems() {
```

(plus imports)

?

File: core/src/main/resources/jenkins/model/Jenkins/projectRelationship.properties
Comment (daniel-beck) [inline core/src/main/resources/jenkins/model/Jenkins/projectRelationship.properties:3]: Seems out of place, this is the English file?

File: core/src/main/resources/jenkins/model/Jenkins/projectRelationship.jelly
Comment (daniel-beck) [inline core/src/main/resources/jenkins/model/Jenkins/projectRelationship.jelly:47]:
```suggestion
          <f:combobox id="lhs" fillUrl="fillJobNameItems" name="lhs" value="${request.getParameter('lhs')}"/>
```

core-pr-tester fails to render these (they're in the page source verbatim), my guess is file system case sensitivity.

File: core/src/main/resources/jenkins/model/Jenkins/projectRelationship.jelly
Comment (daniel-beck) [inline core/src/main/resources/jenkins/model/Jenkins/projectRelationship.jelly:52]:
```suggestion
          <f:combobox id="rhs" fillUrl="fillJobNameItems" name="rhs" value="${request.getParameter('rhs')}"/>
```

File: core/src/main/resources/jenkins/model/Jenkins/projectRelationship.properties
Comment (timja) [reply core/src/main/resources/jenkins/model/Jenkins/projectRelationship.properties:3]: whyyyy is that there, I just copied the header from that file the only one with that -.-

**Review by janfaracik — APPROVED** [review]

**Comment by basil** [comment]
I liked the infix operator used on the old page because it usually denotes a [causal relationship](https://en.wikipedia.org/wiki/Material_conditional) which is the case between an upstream project and downstream project. Surely the fact that GitHub uses this symbol in their [compare view](https://github.com/octocat/linguist/compare/master...octocat:master) between upstream and downstream repositories/branches is evidence that it is sufficiently modern.

**Comment by timja** [comment]
Yes although it doesn't really fit with the maintained form components. I doubt this page is used much and don't think special casing it is worth it. 

The primary motivation for this PR is to remove the long deprecated component.

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after the branch is unlocked, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
